File: nomad-samples/python/basic-live-channel-1.1/live_input/get_live_inputs.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

from libraries import json, requests

'''
 * Get Live Inputs
 *
 * Note: Live Inputs are direct calls,
 *       they are not indexed, thus, the Search API cannot be used.
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @returns JSON Object containing a list of live inputs
'''
async def get_live_inputs(AUTH_TOKEN):
    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Get Live Inputs: Invalid API call")


    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(SERVER_URL + "/liveInput", headers= HEADERS)
    
        # Return JSON response
        return json.loads(RESPONSE.text)

    except:
        await api_exception_handler(RESPONSE, "Get Live Inputs failed")

File: nomad-samples/admin/js/advanced-live-channel-1.1/live-input/get-live-inputs.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Get Live Inputs
 *
 * Note: Live Inputs are direct calls,
 *       they are not indexed, thus, the Search API cannot be used.
 *
 * @param {string} authToken    | Authorization token
 * @returns JSON Object containing a list of live inputs
 */
export default async function getLiveInputs(authToken) {
    // Check for valid parameters
    if (!authToken) {
        throw new Error("Get Live Inputs: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/liveInput`, {
        method: "GET",
        headers: headers
    });

    // Check for success
    if (response && response.ok) {
        // Return JSON response
        return await response.json();
    }

    await apiExceptionHandler(response, "Get Live Inputs failed");
}
